pkg/po/lex.go

```diff
@@ -187,26 +191,49 @@ func parseToken(line string, tokenType TokenType) (*Token, error) {
 
 	token.Index = index
 
-	return &token, nil
+	return token, nil
 }
 
-// parseValue parses the value of the token.
+// parseTokenValue parses the value of the token.
 // e.g.
 //
-//	"Language: en-US\n" -> en-US\n
-//	msgid "some text" -> some text
-//	#, python-format -> python-format
-//	#: superset/key_value/exceptions.py:54 -> superset/key_value/exceptions.py:54
-func parseValue(line string) string {
+//	`"Language: en-US\n"` -> `en-US\n`
+//	`msgid "some text"` -> `some text`
+//	`#, python-format` -> `python-format`
+//	`#: superset/key_value/exceptions.py:54` -> `superset/key_value/exceptions.py:54`
+func parseTokenValue(tokenType TokenType, line string) (string, error) {
 	n := 2
 	fields := strings.SplitN(line, " ", n) // prefix and value, e.g. fields[0] = msgid, fields[1] = "text", etc.
 
-	// No value, only prefix
-	if len(fields) != n {
-		return ""
+	switch tokenType {
+	case TokenTypeMsgID, TokenTypePluralMsgID, TokenTypeMsgStr, TokenTypePluralMsgStr, TokenTypeMsgCtxt:
+		if len(fields) == n && strings.HasPrefix(fields[1], "\"") && strings.HasSuffix(fields[1], "\"") {
+			return strings.TrimPrefix(strings.TrimSuffix(fields[1], "\""), "\""), nil
+		}
+
+		return "", fmt.Errorf("invalid syntax in line string for token type: '%d'", tokenType)
+	case TokenTypeHeaderLanguage, TokenTypeHeaderTranslator, TokenTypeHeaderPluralForms,
+		TokenTypeHeaderProjectIDVersion, TokenTypeHeaderPOTCreationDate, TokenTypeHeaderPORevisionDate,
+		TokenTypeHeaderLanguageTeam, TokenTypeHeaderLastTranslator, TokenTypeHeaderXGenerator,
+		TokenTypeHeaderReportMsgidBugsTo, TokenTypeHeaderMIMEVersion, TokenTypeHeaderContentType,
+		TokenTypeHeaderContentTransferEncoding, TokenTypeHeaderGeneratedBy:
+		if len(fields) == n {
```

Comment:
What is the case when `len(fields) != n`? Why do you ignore it?

Reply:
From the following cases which ones are valid syntax?

```po
"Language:en-US\n"
" Language: en-US\n"
"Language: en-US \n"
```

Reply:
if len(fields) != n empty string is returned.

Reply:
Does `po` allow 1 or more whitespaces? Which cases have valid syntax?

```po
msgid "some text" <-- return `some text`
msgid  "some text" <-- returns empty string in your case
msgid "some text"
msgid"some text"
 msgid "some text"
```

Reply:
ChatGPT:

msgid "some text": This is valid syntax. The keyword msgid is followed by a space and then the quoted string, which is the standard format.

msgid "some text": This is also valid syntax, although it's not standard. There are two spaces between msgid and the quoted string, but most parsers will handle this correctly. However, for consistency and readability, it's generally recommended to use only one space.

msgid"some text": This is not valid syntax. There should be a space between msgid and the quoted string. Without this space, the line may not be correctly parsed.

msgid "some text": Technically, this is valid syntax. The parser usually ignores leading whitespace, including the space before msgid. However, it's not standard or recommended practice to include this leading space, as it can affect readability and consistency in the file.

In summary, the first two lines are valid, with the second being non-standard due to the extra space. The third line is invalid due to the lack of space between msgid and the quote. The fourth line, while technically valid, is non-standard due to the leading space. For best practices, it's recommended to follow the format shown in the first line.

Reply:
okay will add some tests

Reply:
@klavinsjanis what does po spec says about it?

Reply:
Is there formal specification?